ion-schema/src/system.rs

```diff
@@ -931,30 +1039,57 @@ impl Resolver {
                     _ => Err(error),
                 },
                 Ok(schema_content) => {
-                    let isl = self.isl_schema_from_elements(schema_content.into_iter(), id)?;
-                    self.schema_from_isl_schema(isl, id, type_store, load_isl_import)
+                    // ISL version marker regex
+                    let isl_version_marker = Regex::new(r"^\$ion_schema_\d.*$").unwrap();
```

Comment:
Compiling a `Regex` can be expensive. Can you move this out of the `authorities` loop?